mobile/src/test/java/org/openhab/habdroid/model/ParsedStateTest.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.habdroid.model
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class ParsedStateTest {
+    @Test
+    fun testParseAsBoolean() {
+        listOf("ON", "100", "1", "0.1").forEach {
+            assertTrue("State $it should be parsed as true", ParsedState.parseAsBoolean(it))
+        }
+        listOf("OFF", "0").forEach {
+            assertFalse("State $it should be parsed as false", ParsedState.parseAsBoolean(it))
+        }
+    }
+
+    @Test
+    fun testParseAsNumber() {
+        mapOf(
+            "ON" to 100f,
+            "OFF" to 0f,
+            "3" to 3f,
+            "0" to 0f,
+            "42.42" to 42.42f
+        ).forEach {
+            val expected = ParsedState.NumberState(it.value)
+            val actual = ParsedState.parseAsNumber(it.key, null)
+            assertEquals("${it.key} should be parsed as ${it.value}", expected, actual)
+        }
+    }
+
+    @Test
+    fun testParseAsBrightness() {
+        mapOf(
+            "100" to 100,
+            "0" to 0,
+            "0.1" to 0, // TODO
```

Comment:
Not sure about this one. A light bulb with `0.1` brightness isn't off, so maybe parse everything between 0 and 1 as brightness 1?